app/controllers/smart_proxies_controller.rb

```diff
@@ -23,16 +27,10 @@ def edit
     @proxy = @smart_proxy
   end
 
-  def ping
-    @proxy = @smart_proxy
-    respond_to do |format|
-      format.json {render :json => errors_hash(@smart_proxy.refresh)}
-    end
-  end
-
   def refresh
     old_features = @smart_proxy.features
     if @smart_proxy.refresh.blank? && @smart_proxy.save
+      delete_cached_versions
       msg = @smart_proxy.features == old_features ? _("No changes found when refreshing features from %s.") : _("Successfully refreshed features from %s.")
```

Comment:
to fix this, change line 31 and here to use `@smart_proxy.features.to_a` which will force a query instead of lazy relation that is evaluated on comparison (and obviously gives the same result)